apps/labs/posts/pep-517-build-system-popularity.md

```diff
@@ -0,0 +1,615 @@
+code, and I counted all the packages containing `setup.py` as using it.
+However, some packages used it only as a compatibility script without
+actually declaring any metadata in it. `setup.cfg` and `pyproject.toml`
+were counted if they actually contained a package metadata section.
+
+## Build system dependencies
+
+PEP 517 specifies that the packages that need to be installed for the build
+process come from two sources:
+
+- the values declared in `build-system.requires` key of `pyproject.toml`, and
+
+- the values returned by `get_requires_for_build_wheel()` function
+  of the PEP 517 build backend.
+
+The packages listed in `pyproject.toml` are installed first. It often lists
+all build requirements for the project, but it must at least include all
+packages that are required to invoke the build backend (e.g. `setuptools`
+and all packages imported in `setup.py`).
+
+The packages provided by the backend function usually include the dependencies
+of the build backend itself, and dynamic dependencies of the project.
+For example, the scikit-build-core backend uses it to request installing `cmake`
+and `ninja` packages if the respective tools are not present on the system.
+
+Out of 7434 source distributions analyzed, 8 used top-level directories
+that did not match their filenames. Some of them used normalized
+directory names but non-normalized filenames (e.g. `pyre-check-0.9.23.tar.gz`
+contained `pyre_check-0.9.23` directory), others the other way around
+(`PyICU-2.14.tar.gz` contained `pyicu-2.14`). A few packages used even
+more arbitrary directories, e.g. `Distance-0.1.3.tar.gz` used `distance`,
+while `kuzu-0.7.1.tar.gz` used `sdist`.
+
+306 source distributions raised an exception while calling their
+`get_requires_for_build_wheel()` function. This means that these
+distributions could not be installed on my system. Out of these:
+
+- 95 belonged to `pyobjc-framework` that supports macOS only
+
+- 19 failed due to missing system packages that cannot be installed
+  from PyPI
+
+- two represented deprecated package aliases that weren't meant to be
+  installed, and only inform the user to use another package
+
+- one was a backport that supports Python 2 only
+
+- one failed while building a C extension (it shouldn't have started
+  building anything yet)
+
+I discounted these packages as having special requirements, and focused
+on the remaining 188 source distributions. These clearly failed because
+of bugs. This number included:
+
+- 90 packages missing some required files (such as README or requirement
+  files specified in `setup.py`)
+
+- 64 missing Python dependencies in `build-system.requires` key
+
+The remaining packages were either failing due to incorrect metadata,
+use of removed setuptools API, code incompatible with Python 3.11
+or expecting being built from a git checkout.
+
+<table id='table-5' style={{margin: 'auto', width: 'auto'}}>
+  <caption>Table 5. Setuptools and wheel dependencies</caption>
```

Comment:
No change to the blog post needed, but: I'd be curious to learn which packages were the `1` for scikit-build-core and for meson-python.

Reply:
scikit-image for mesonpy, soxr for scikit-build-core. Both seem unused.

Technically, [scikit-build-core has a setuptools variant](https://scikit-build-core.readthedocs.io/en/latest/plugins/setuptools.html) but I didn't see that used in the wild (yet).